Sources/Recycling/ListViews/FavoriteFolders/FavoriteFoldersListView.swift

```diff
@@ -0,0 +1,296 @@
+        tableView.backgroundColor = .milk
+        tableView.rowHeight = FavoriteFoldersListView.estimatedRowHeight
+        tableView.estimatedRowHeight = FavoriteFoldersListView.estimatedRowHeight
+        tableView.separatorInset = .leadingInset(frame.width)
+        tableView.tableFooterView = UIView()
+        tableView.contentInset.bottom = FavoriteFoldersListView.estimatedRowHeight
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(AddFavoriteFolderViewCell.self)
+        tableView.register(RemoteImageTableViewCell.self)
+        return tableView
+    }()
+
+    private lazy var footerView: FavoriteFoldersFooterView = {
+        let view = FavoriteFoldersFooterView(withAutoLayout: true)
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var footerViewTop = footerView.topAnchor.constraint(equalTo: bottomAnchor)
+
+    private lazy var footerHeight: CGFloat = {
+        return 56 + windowSafeAreaInsets.bottom
+    }()
+
+    // MARK: - Init
+
+    public required init(viewModel: FavoriteFoldersListViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Reload
+
+    public func reloadData() {
+        tableView.reloadData()
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        searchBar.configure(withPlaceholder: viewModel.searchBarPlaceholder)
+        footerView.configure(withTitle: viewModel.addFolderText)
+
+        let barButtonAppearance = UIBarButtonItem.appearance(whenContainedInInstancesOf: [FavoriteFoldersSearchBar.self])
+        barButtonAppearance.title = viewModel.cancelButtonTitle
+
+        addSubview(tableView)
+        addSubview(searchBar)
+        addSubview(footerView)
+
+        NSLayoutConstraint.activate([
+            searchBar.topAnchor.constraint(equalTo: topAnchor),
+            searchBar.leadingAnchor.constraint(equalTo: leadingAnchor),
+            searchBar.trailingAnchor.constraint(equalTo: trailingAnchor),
+
+            tableView.topAnchor.constraint(equalTo: searchBar.bottomAnchor),
+            tableView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: bottomAnchor),
+
+            footerViewTop,
+            footerView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            footerView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            footerView.heightAnchor.constraint(equalToConstant: footerHeight)
+        ])
+    }
+}
+
+// MARK: - UITableViewDataSource
+
+extension FavoriteFoldersListView: UITableViewDataSource {
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        return Section.allCases.count
```

Comment:
Never thought of handling multiple sections in a Table/Collection view this way before, that's really clever 👍 👍